src/puppetlabs/trapperkeeper/authorization/acl.clj

```diff
@@ -79,30 +76,73 @@
   [pattern]
   (-> pattern (str/lower-case) (str/split #"\.") reverse vec))
 
-(def allow-all { :auth-type :allow :type :allow-all :qualifier :exact :length nil :pattern [] })
+(def allow-all {:auth-type :allow
+                :type :allow-all
+                :qualifier :exact
+                :length nil
+                :pattern []})
+
+(def deny-all {:auth-type :deny
+               :type :regex
+               :qualifier :exact
+               :length nil
+               :pattern #"^/.*/$"})
```

Comment:
I don't think this pattern is correct.  I think we want `#"^.*$"` or just `#"^*$"`.  With what you have, the `match-name?` function will return `false` when checking to see if this ACE matches a request name except for cases where the name is delimited with a forward-slash and back-slash.  So `/blah/` or just `//` would return `true` for `match-name?` but `blah` would return `false`.  For a wildcard like this, I think we want `match-name?` to return `true` regardless of the value in the name.

To validate this in a unit test, for example, try adding an allow ACE to the rule being used the new "Denied when deny all" test and note that if the request would end up being allowed if the name were delimited with a forward slash and back slash but denied otherwise.